salt/docker/init.sls

```diff
@@ -16,6 +16,7 @@ include:
         Environment="NO_PROXY={{ salt['pillar.get']('proxy:no_proxy', '') }}"
   cmd.run:
     - name: systemctl daemon-reload
+    - stateful: True
```

Comment:
So, I think @inercia is right here... Given how salt orders things, this could happen (in theory):

1) We edit the file
2) Docker is restarted - old proxy info is still in use
3) Systemd is reloaded

when what we really want is:

1) We edit the file
2) Systemd is reloaded
3) Docker is restarted - new proxy info is now in use